torch/fx/experimental/validator.py

```diff
@@ -277,12 +277,42 @@ def wrapper(*args):
 
             return wrapper
 
+        # Like `lift`, but for bitwise ops. We need to convert to/from BitVec
+        # in order to use z3 bitwise ops. We assume that integers are 64 bit.
+        def bitwise_lift(func):
+            def wrap(a) -> z3.ExprRef:
+                if isinstance(a, bool):
+                    a = z3.BoolVal(a)
+                if isinstance(a, (int, sympy.Integer)):
+                    a = z3.IntVal(int(a))
+                if not isinstance(a, (z3.ArithRef, z3.BoolRef)):
+                    raise ValueError(f"can't lift type for bitwise op: {type(a)}")
+                try:
+                    return z3.Int2BV(a, 64)
+                except z3.Z3Exception:
+                    # Catch z3 Int/Bool sort error
+                    return z3.Int2BV(a != 0, 64)
+
+            @functools.wraps(func)
+            def wrapper(*args):
+                # Lifts the arguments into a list of Z3 inhabitants.
+                if len(args) == 1 and isinstance(args[0], (list, tuple)):
+                    wrapped_args = (tuple(wrap(a) for a in args[0]),)
+                else:
+                    wrapped_args = tuple(wrap(a) for a in args)
+                # Run the function on the Z3 expressions.
+                return z3.BV2Int(func(*wrapped_args))
+
+            return wrapper
+
         ops = _Z3Ops(validator)
         replacement_map = {
             # Operator module.
             operator.not_: lift(z3.Not),
-            operator.and_: lift(z3.And),
-            operator.or_: lift(z3.Or),
+            operator.and_: bitwise_lift(operator.and_),
```

Comment:
Maybe for operators `and_` and `or_` we could check if the arguments are bool, in order to make it easier for Z3 to solve, instead of doing logical operations in BV.

Reply:
I can try having a separate case when both arguments are bool.

Reply:
Yeah, I am worried that the change here is going to cause problems for regular bool conjunction/disjunction. Ideally, we are able to distinguish if something is a logical and / logical or versus bitwise and / bitwise or. If we've been abusing `operator.and_` and `operator.or_` for both cases, we may need to add a dedicated logical and/or that is its dedicated operator.

This is especially pernicious because bool is a subtype of int and they are NOT compatible.